### PR TITLE
feat(sdk): Add token list to sdk

### PR DESCRIPTION
**Pull request by roninjin10**
This adds cloning the token list as a build step.  

**Why**
- We want to use token list as source of truth instead of duplicating things like token address constants and bridge adapters https://docs.google.com/document/d/1oGoXwKIh3YD3BdviT3OuHILG0iZSDRTB31eut22Z1WI/edit?usp=sharing
- Versioning the token list makes our build more hermetic on the frontend.   We previously have had the UI break out from under us.

I held off on updating the adapters for now because that is blocked by this pr adding all the existing adapters https://github.com/ethereum-optimism/ethereum-optimism.github.io/pull/186

### CODE REVIEW

**Comment by nickbalestra** [comment]
What are the benefits of fetching it over adding it as a git submodule during the build step?

**Comment by roninjin10** [comment]
@nickbalestra 

1. More convenient.  For this the token list is always up to date without having to change a hash
2. @smartcontracts would not approve the pr if I used git submodules because he thinks they are gross
3. I don't like them either

The downside is the build still isn't hermetic with this change.   We can make it hermetic though later by publishing the token list as an npm package directly and having it as a sub dependency here.

File: packages/sdk/src/utils/token-list-utils.ts
Comment (nickbalestra) [inline packages/sdk/src/utils/token-list-utils.ts:34]: Just to make sure I am not missing something, what is the reason for having those options if are not needed/used in this PR?

File: packages/sdk/scripts/generateTokenList.ts
Comment (nickbalestra) [inline packages/sdk/scripts/generateTokenList.ts:6]: should this be in `/misc-constants` ?

**Review by nickbalestra — CHANGES_REQUESTED** [review]
I would suggest to avoid adding options we don't need yet (see comment) 
Curious on what made you go with fetching vs adding git module during build step

File: packages/sdk/src/utils/token-list-utils.ts
Comment (roninjin10) [reply packages/sdk/src/utils/token-list-utils.ts:34]: They are going to be used in the gateway that's where this came from.  I'm exporting this as a common util for anybody to use 
image

File: packages/sdk/scripts/generateTokenList.ts
Comment (roninjin10) [reply packages/sdk/scripts/generateTokenList.ts:6]: misc-constants imports it from src/generated

**Review by nickbalestra — APPROVED** [review]

**Comment by nickbalestra** [comment]
Thanks for the clarifications @roninjin10! (Btw, I'm pretty aligned with @smartcontracts on this :) - Approved

File: packages/sdk/package.json
Comment (roninjin10) [inline packages/sdk/package.json:12]: The indexer in future will use the standard token list to find custom bridges it needs to index

File: packages/sdk/package.json
Comment (tynes) [inline packages/sdk/package.json:14]: I don't think you need to call `rimraf` twice here

File: packages/sdk/src/utils/misc-constants.ts
Comment (tynes) [inline packages/sdk/src/utils/misc-constants.ts:2]: I don't love that there are dots in the filename here

**Comment by roninjin10** [comment]
Going to change this to make sdk not hardcode any tokens.   Instead it will take tokens as a constructor arg

File: packages/sdk/package.json
Comment (roninjin10) [reply packages/sdk/package.json:14]: Good catch

**Pull request closed without merging**
